vendor/bat-native-ads/src/bat/ads/internal/ml/pipeline/text_processing/text_processing_unittest.cc

```diff
@@ -0,0 +1,235 @@
+    prediction_maps[i] = prediction_map;
+  }
+
+  // Assert
+  for (size_t i = 0; i < prediction_maps.size(); i++) {
+    const PredictionMap& prediction_map = prediction_maps[i];
+    for (const auto& pred : prediction_map) {
+      const double other_prediction = pred.second;
+      EXPECT_TRUE(prediction_map.at(train_labels[i]) >= other_prediction);
+    }
+  }
+}
+
+TEST_F(BatAdsTextProcessingPipelineTest, InitValidModelTest) {
+  // Arrange
+  pipeline::TextProcessing text_proc_pipeline;
+  const base::Optional<std::string> opt_value =
+      ReadFileFromTestPathToString(kValidSegmentClassificationPipeline);
+
+  // Act
+  ASSERT_TRUE(opt_value.has_value());
+  const std::string model_json = opt_value.value();
+  bool loaded_successfully = text_proc_pipeline.FromJson(model_json);
+
+  // Assert
+  EXPECT_TRUE(loaded_successfully);
+}
+
+TEST_F(BatAdsTextProcessingPipelineTest, InvalidModelTest) {
+  // Arrange
+  pipeline::TextProcessing text_proc_pipeline;
+  const base::Optional<std::string> opt_value =
+      ReadFileFromTestPathToString(kInvalidSpamClassificationPipeline);
+
+  // Act
+  ASSERT_TRUE(opt_value.has_value());
+  const std::string model_json = opt_value.value();
+  bool loaded_successfully = text_proc_pipeline.FromJson(model_json);
+
+  // Assert
+  EXPECT_FALSE(loaded_successfully);
+}
+
+TEST_F(BatAdsTextProcessingPipelineTest, EmptyModelTest) {
+  // Arrange
+  pipeline::TextProcessing text_proc_pipeline;
+  const std::string empty_model_json = "{}";
+
+  // Act
+  bool loaded_successfully = text_proc_pipeline.FromJson(empty_model_json);
+
+  // Assert
+  EXPECT_FALSE(loaded_successfully);
+}
+
+TEST_F(BatAdsTextProcessingPipelineTest, MissingModelTest) {
+  // Arrange
+  pipeline::TextProcessing text_proc_pipeline;
+
+  // Act
+  const std::string missing_model_json = "";
+  bool loaded_successfully = text_proc_pipeline.FromJson(missing_model_json);
+
+  // Assert
+  EXPECT_FALSE(loaded_successfully);
+}
+
+TEST_F(BatAdsTextProcessingPipelineTest, TopPredUnitTest) {
+  // Arrange
+  const size_t kMaxPredictionsSize = 100;
+  const std::string kTestPage = "ethereum bitcoin bat zcash crypto tokens!";
+  pipeline::TextProcessing text_proc_pipeline;
+  const base::Optional<std::string> opt_value =
+      ReadFileFromTestPathToString(kValidSegmentClassificationPipeline);
+
+  // Act
+  ASSERT_TRUE(opt_value.has_value());
+  const std::string model_json = opt_value.value();
+  ASSERT_TRUE(text_proc_pipeline.FromJson(model_json));
+  const PredictionMap preds = text_proc_pipeline.ClassifyPage(kTestPage);
+
+  // Assert
+  ASSERT_TRUE(preds.size());
+  ASSERT_LT(preds.size(), kMaxPredictionsSize);
+  ASSERT_TRUE(preds.count("crypto-crypto"));
+  for (auto const& pred : preds) {
+    EXPECT_TRUE(pred.second <= preds.at("crypto-crypto"));
+  }
+}
+
+TEST_F(BatAdsTextProcessingPipelineTest, TextCMCCrashTest) {
+  // Arrange
+  const size_t kMinPredictionsSize = 2;
+  const size_t kMaxPredictionsSize = 100;
```

Comment:
Not used, please check elsewhere and remove anything that is not used. Thanks

Reply:
Both are used in assertions (`ASSERT_GT`, `ASSERT_LT`).